text/2020-01-30-key-spaces.md

```diff
@@ -0,0 +1,181 @@
+# Title
+
+Key Spaces
+
+## Summary
+
+When multiple applications store data in TiKV, we need a clear pattern for avoiding conflicting use of the key space in TiKV.
+TiKV is a monolithic KV store, so each application must choose a key prefix.
+Key Spaces formalizes this process by registering a key space for an application in TiKV itself.
+
+When combined with authentication (not part of this proposal), this can secure an application's key space.
+
+Additionally, key spaces make it possible for TiKV to avoid conflicts during a data restore by changing the key prefix assigned to a key space.
+
+## Motivation
+
+The use case for this is running multiple applications against TiKV (not just a single TiDB).
+Note that we may be motivated to run multiple instances of a single application such as TiDB.
+
+* Avoiding even accidental conflicting use of key space in TiKV is currently ad-hoc and unreliable
+* Securing data in TiKV is difficult. A necessary first step is clearly defining the boundaries of data.
```

Comment:
I think we can define the boudaries now. What is the defficulty?

Reply:
Yes you can define boundaries by writing them down on a piece of paper way but how do you keep it up to date when new applications are created and prevent someone from accessing the key space that belongs to your application in an automated way?